cylc/flow/workflow_files.py

```diff
@@ -1759,23 +1753,50 @@ def get_run_dir_info(
     return relink, run_num, rundir
 
 
-def detect_both_flow_and_suite(path):
+def detect_both_flow_and_suite(path: Path) -> None:
     """Detects if both suite.rc and flow.cylc are in directory.
+
+    Permits flow.cylc to be a symlink.
+    Return true if present, raises error if flow.cylc path sent is a forbidden
+    symlink.
     Raises:
         WorkflowFilesError: If both flow.cylc and suite.rc are in directory
     """
+    flow_cylc = None
     msg = (f"Both {WorkflowFiles.FLOW_FILE} and {WorkflowFiles.SUITE_RC} "
-           "files are present in the run directory. Please remove one and"
+           f"files are present in {path}. Please remove one and"
            " try again. For more information visit: https://cylc.github.io/"
            "cylc-doc/latest/html/7-to-8/summary.html#backward-compatibility")
     if path.resolve().name == WorkflowFiles.SUITE_RC:
         flow_cylc = path.parent / WorkflowFiles.FLOW_FILE
-        if flow_cylc.is_file() and not flow_cylc.is_symlink():
-            raise WorkflowFilesError(msg)
     elif (path / WorkflowFiles.SUITE_RC).is_file():
         flow_cylc = path / WorkflowFiles.FLOW_FILE
-        if flow_cylc.is_file() and not flow_cylc.is_symlink():
-            raise WorkflowFilesError(msg)
+    if flow_cylc and flow_cylc.is_file() and is_forbidden(flow_cylc):
+        raise WorkflowFilesError(msg)
+
+
+def is_forbidden(flow_file: Path) -> bool:
+    """Returns True for a forbidden file structure scenario.
+
+    Forbidden criteria:
+        A symlink elsewhere on file system but suite.rc also exists in the
+        directory.
+        flow.cylc and suite.rc in same directory but no symlink
+    Args:
+        flow_file : Absolute Path to the flow.cylc file
+    """
+    if not flow_file.is_symlink():
+        if flow_file.parent.joinpath(WorkflowFiles.SUITE_RC).exists():
+            return True
+        return False
+    link = flow_file.resolve()
+    if link.parent == flow_file.parent:
```

Comment:
This would allow
```
myflow
|-- something.cylc
|-- flow.cylc -> something.cylc
`-- suite.rc
```
which is equivalent to having distinct `flow.cylc` and `suite.rc` files in the same run dir.

Perhaps it should be something like
```python
if flow_file.resolve() == flow_file.parent / WorkflowFiles.SUITE_RC:
```

Reply:
This case (I think) is covered by the following condition: `if flow_file.parent.joinpath(WorkflowFiles.SUITE_RC).exists():`.

Reply:
It would be, but it will `return False` before getting to that

Reply:
Sorry for my confusion! All testing for me was raising the error....
![image](https://user-images.githubusercontent.com/37735232/144404626-1c909b8b-7f71-4b51-aca7-567fee807ff2.png)
The return False does not return before that (where it should!) because `flow_file.parent` needed a `resolve()` on the end to hit the false.
Will go with your suggestion (adding a resolve()), thanks @MetRonnie

Reply:
Ok, coming back to this (confusion after your suggestion was added by Tim so it wasn't visible to me). This suggestion will forbid symlinking elsewhere. 
Going to push a change which I think now covers all scenarios, based on your `if flow_file.resolve() == flow_file.parent / WorkflowFiles.SUITE_RC:` suggestion.